src/cli/cli.cpp

```diff
@@ -602,6 +612,259 @@ void Interpreter::ProcessChildTimeout(int argc, char *argv[])
     AppendResult(error);
 }
 
+#if OPENTHREAD_ENABLE_APPLICATION_COAP
+
+void Interpreter::ProcessCoapServer(int argc, char *argv[])
+{
+    ThreadError error = kThreadError_None;
+
+    VerifyOrExit(argc > 0, error = kThreadError_InvalidArgs);
+
+    if (strcmp(argv[0], "start") == 0)
+    {
+        SuccessOrExit(error = otCoapServerStart(mInstance));
+        SuccessOrExit(error = otCoapServerAddResource(mInstance, &mResource));
+        sServer->OutputFormat("Starting server and adding resource '%s' ... ", mResource.mUriPath);
+    }
+    else if (strcmp(argv[0], "stop") == 0)
+    {
+        otCoapServerRemoveResource(mInstance, &mResource);
+        SuccessOrExit(error = otCoapServerStop(mInstance));
+        sServer->OutputFormat("Stopping server ... ");
+    }
+    else if (strcmp(argv[0], "name") == 0)
+    {
+        if (argv[1]){
+            mResource.mUriPath = argv[1];
+            sServer->OutputFormat("Changing server resource to '%s' ... ", argv[1]);
+        }
+        else
+        {
+            ExitNow(error = kThreadError_Parse);
+        }
+    }
+    else
+    {
+        ExitNow(error = kThreadError_InvalidArgs);
+    }
+
+exit:
+    (void)argc;
+    (void)argv;
+    AppendResult(error);
+}
+
+void OTCALL Interpreter::s_HandleCoapServerResponse(void *aContext, otCoapHeader *aHeader, otMessage *aMessage, otMessageInfo *aMessageInfo)
+{
+    static_cast<Interpreter *>(aContext)->HandleCoapServerResponse(aHeader, aMessage, aMessageInfo);
+}
+
+void Interpreter::HandleCoapServerResponse(otCoapHeader *aHeader, otMessage *aMessage, otMessageInfo *aMessageInfo)
+{
+    sServer->OutputFormat("Received CoAP request from [%x:%x:%x:%x:%x:%x:%x:%x]: ",
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[0]),
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[1]),
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[2]),
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[3]),
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[4]),
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[5]),
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[6]),
+							  HostSwap16(aMessageInfo->mSockAddr.mFields.m16[7]));
+
+    switch (otCoapHeaderGetCode(aHeader))
+    {
+        case kCoapRequestGet:
+            sServer->OutputFormat("GET");
+            break;
+        case kCoapRequestDelete:
+            sServer->OutputFormat("DELETE");
+            break;
+        case kCoapRequestPut:
+            sServer->OutputFormat("PUT");
+            break;
+        case kCoapRequestPost:
+            sServer->OutputFormat("POST");
+            break;
+        default:
+            sServer->OutputFormat("Undefined\r\n");
+            return;
+    }
+
+    // Output payload of message if existent
+    uint16_t payloadLength = otMessageGetLength(aMessage) - otMessageGetOffset(aMessage);
+    if (payloadLength > 0)
+    {
+        char payload[payloadLength] = {};
```

Comment:
Sure, i was thinking about something like this:
https://github.com/openthread/openthread/blob/master/src/core/common/message.cpp#L670
https://github.com/openthread/openthread/blob/master/src/cli/cli.cpp#L2980